pex/environment.py

```diff
@@ -307,31 +307,40 @@ def _resolve(self, working_set, reqs):
     return resolveds
 
   @staticmethod
-  def _declare_namespace_packages(dist):
-    if not dist.has_metadata('namespace_packages.txt'):
+  def _declare_namespace_packages(resolved_dists):
+    namespace_package_dists = [dist for dist in resolved_dists
+                               if dist.has_metadata('namespace_packages.txt')]
+    if not namespace_package_dists:
       return  # Nothing to do here.
 
-    # NB: Some distributions (notably `twitter.common.*`) in the wild declare setuptools-specific
-    # `namespace_packages` but do not properly declare a dependency on setuptools which they must
+    # NB: Some distributions (notably `twitter.common.*`) in the wild declare `setuptools`-specific
+    # `namespace_packages` but do not properly declare a dependency on `setuptools` which they must
     # use to:
-    # 1. Declare namespace_packages metadata which we just verified they have with the check above.
+    # 1. Declare `namespace_packages` metadata which we just verified they have with the check
+    #    above.
     # 2. Declare namespace packages at runtime via the canonical:
     #    `__import__('pkg_resources').declare_namespace(__name__)`
     #
-    # As such, we assume the best and try to import pkg_resources from the distribution they depend
-    # on, and then fall back to our vendored version only if not present. This is safe, since we'll
-    # only introduce our shaded version when no other standard version is present and even then tear
-    # it all down when we hand off from the bootstrap to user code.
+    # As such, we assume the best and try to import `pkg_resources` from the distribution they
+    # depend on (1), and then fall back to our vendored version only if not present. This is safe,
```

Comment:
The `(1)` took me a second to figure out what it meant. I recommend using `*` instead, although fine to keep as is.

Reply:
Heh, that's where I started and it looked weird to me so I went htis route (I really wanted dagger). I'll re-structure to use neither but treat both branches - we supply setuptools for bad actors and we use their setuptools - as equals / bullet points.